### PR TITLE
Bugfix for wget2 null status description.

### PR DESCRIPTION
**Pull request by nhhollander**
Fixes an issue where the `emscripten_async_wget2_data` `onerror` callback would always have a null status description.

See #14373

Creating a draft PR while I work on test cases.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -567,4 +567,5 @@ a license to everyone to use it as detailed in LICENSE.)
 * Zhi An Ng <zhin@google.com> (copyright owned by Google, Inc.)
 * Ian Jackson <ijackson@chiark.greenend.org.uk>
 * Brian Gontowski <brian@gontowski.com>
-* Bobbie Chen <bobbie.chen75@gmail.com>
\ No newline at end of file
+* Bobbie Chen <bobbie.chen75@gmail.com>
+* Nicholas Hollander <nhollander98@gmail.com>
```
Comment (sbc100) [inline AUTHORS:571]: Can you add a newline at the end?

File: src/library_browser.js [hunk 2]
```diff
@@ -977,15 +977,29 @@ var LibraryBrowser = {
         if (onload) {{{ makeDynCall('viiii', 'onload') }}}(handle, arg, buffer, byteArray.length);
         if (free) _free(buffer);
       } else {
-        if (onerror) {{{ makeDynCall('viiii', 'onerror') }}}(handle, arg, http.status, http.statusText);
+        if (onerror) {
+          var statusText = 0;
+          if(http.statusText) {
```
Comment (sbc100) [inline src/library_browser.js:982]: Space after `if`

File: src/library_browser.js [hunk 3]
```diff
@@ -977,15 +977,29 @@ var LibraryBrowser = {
...
+            var len = lengthBytesUTF8(http.statusText) + 1;
+            statusText = stackAlloc(len);
+            stringToUTF8(http.statusText, statusText, len);
```
Comment (sbc100) [inline src/library_browser.js:985]: @kripken do we have any shorthand for this?  Seems like it would be a fairly common pattern.

**Review by sbc100 — APPROVED** [review]
thanks!

**Comment by nhhollander** [comment]
The test cases use http://httpstat.us/ to generate response codes and CORS errors for testing, not sure if that's an acceptable dependency. Maybe there's a way to generate or emulate those internally from the test system?

File: src/library_browser.js [hunk 4]
```diff
@@ -977,15 +977,29 @@ var LibraryBrowser = {
         if (onload) {{{ makeDynCall('viiii', 'onload') }}}(handle, arg, buffer, byteArray.length);
         if (free) _free(buffer);
       } else {
-        if (onerror) {{{ makeDynCall('viiii', 'onerror') }}}(handle, arg, http.status, http.statusText);
+        if (onerror) {
```
[the remark this answers is not in the document]
Comment (nhhollander) [reply src/library_browser.js:980]: It definitely could, although I'm wondering if it should actually be an entirely separate function. The `http.onload` handler for non-2xx codes and the `http.onerror` handler are functionally identical.

[the remark this answers is not in the document]
Comment (sbc100) [reply src/library_browser.js:980]: Yes it looks like you can extract function (perhaps called `onerrorjs` as opposed to the existing `onerror` which is a native function pointer).  That would better for sure.

**Comment by sbc100** [comment]
Also, a word or warning, I have a PR open that move the wget code into a different file:  https://github.com/emscripten-core/emscripten/pull/14388.    One of is will need to handle the conflict... I cant wait for this PR assuming its almost ready?

**Comment by nhhollander** [comment]
I'll move the function and then it should be good to go, assuming there's no other feedback.

**Review by sbc100 — APPROVED** [review]

**Comment by sbc100** [comment]
Looks like you have some conflicts to resolve and then we are good to land.

**Comment by nhhollander** [comment]
I think it's good

**Comment by sbc100** [comment]
I enabled auto-merge so as a soon as tests pass it should land

**Pull request merged**